src/plugins/Plugin.ts

```diff
@@ -15,28 +16,34 @@ export type PluginContext = {
     getSession: GetSession;
 };
 
-export interface Plugin {
+export abstract class Plugin {
+    protected enabled: boolean = true;
+
+    constructor(
+        public readonly pluginName: string,
```

Comment:
question: Do we need both `pluginName` and `pluginId`?

Reply:
I think so, especially if we get to a point where ids could get a bit abstract

Reply:
Semantically, what is the difference between the two? When would we use one over the other?

It seems like a single ID is sufficient for `PluginManager` to uniquely identify the plugin, which I believe is the only place it is used.

Reply:
well in one of my commits, I did rename `pluginId` to `awsRumId` or something like that -- I think (personally) that the aws id should not be the plugin identifier in the code, and we should have a cleaner "id" (here named `pluginName`) to identify the plugin in code.

Reply:
I don't see any use case for `pluginName`. Let's stick with `pluginId` only.